prometheus/indico_prometheus/plugin.py

```diff
@@ -0,0 +1,62 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2002 - 2023 CERN
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from datetime import timedelta
+
+from wtforms.fields import StringField
+from wtforms.validators import DataRequired, Optional
+
+from indico.core.plugins import IndicoPlugin
+from indico.core.settings.converters import TimedeltaConverter
+from indico.web.forms.base import IndicoForm
+from indico.web.forms.fields import TimeDeltaField
+
+from indico_prometheus import _
+from indico_prometheus.blueprint import blueprint
+
+
+class PluginSettingsForm(IndicoForm):
+    cache_ttl = TimeDeltaField(
+        _('Cache TTL'),
+        [DataRequired()],
+        description=_('TTL for cache'),
+        units=('seconds', 'minutes', 'hours')
+    )
+    token = StringField(
```

Comment:
I'd use an `IndicoPasswordField` with `toggle=True` for this, and add validation for the prefix:

```py
    def validate_token(self, field):
        if field.data and not field.data.startswith('inds_metrics_'):
            raise ValidationError(_('...'))
```

Reply:
I thought asking the user to write the prefix would be a bit cumbersome. How do you see that?

Reply:
oh it's handled implicitly. nevermind in that case. in that case the description is kind of confusing though...